pkg/kdk/kubesync.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kdk
+
+import (
+	"github.com/docker/docker/api/types"
+	log "github.com/sirupsen/logrus"
+)
+
+func Kubesync(cfg KdkEnvConfig) {
+
+	// Check if KDK container is running
+	kdkRunning := false
+
+	containers, err := cfg.DockerClient.ContainerList(cfg.Ctx, types.ContainerListOptions{All: true})
+
+	if err != nil {
+		log.WithField("error", err).Fatal("Failed to list docker containers")
+	}
+
+	for _, container := range containers {
```

Comment:
SHOULD: I've seen this code block like 5 times in the KDK.  Would be great to stop duplicating it.